powerstore/datasource_replication_rule.go

```diff
@@ -0,0 +1,268 @@
+	"terraform-provider-powerstore/models"
+
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ datasource.DataSource              = &replicationRuleDataSource{}
+	_ datasource.DataSourceWithConfigure = &replicationRuleDataSource{}
+)
+
+// newReplicationRuleDataSource returns the replication rule data source object
+func newReplicationRuleDataSource() datasource.DataSource {
+	return &replicationRuleDataSource{}
+}
+
+type replicationRuleDataSource struct {
+	client *client.Client
+}
+
+func (d *replicationRuleDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_replication_rule"
+}
+
+func (d *replicationRuleDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description:         "This datasource is used to query the existing replication rule from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		MarkdownDescription: "This datasource is used to query the existing replication rule from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the replication rule. Conflicts with `name`.",
+				MarkdownDescription: "Unique identifier of the replication rule. Conflicts with `name`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Name of the replication rule. Conflicts with `id`.",
+				MarkdownDescription: "Name of the replication rule. Conflicts with `id`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"replication_rules": schema.ListNestedAttribute{
+				Description:         "List of replication rules.",
+				MarkdownDescription: "List of replication rules.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Unique identifier of the replication rule. Conflicts with `name`.",
```

Comment:
This does not conflict with anyone. Same for the `name` attribute below.
```suggestion
							Description:         "Unique identifier of the replication rule.",
```